test/sdk/sdk.test.ts

```diff
@@ -0,0 +1,321 @@
+    console.log('deriveAccountOnTuring; ', deriveAccountOnTuring);
+  
+    // Transfer
+    await astarAdapter.crossChainTransfer(
+      turingAdapter.getLocation(),
+      deriveAccountOnTuring,
+      defaultAsset.location,
+      new BN('200000000000000000'),
+      keyringPair,
+    );
+  }, 1000000);
+
+  afterAll(async () => {
+    await turingApi?.disconnect();
+    await astarApi?.disconnect();
+  }, 1000000);
+});
+
+describe('test-mangata', () => {
+  let keyringPair: KeyringPair | undefined = undefined;
+  let turingApi: ApiPromise | undefined = undefined;
+  let turingAdapter: OakAdapter | undefined = undefined;
+  let mangataSdk: Mangata | undefined = undefined;
+  let mangataApi: ApiPromise | undefined = undefined;
+  let mangataAdapter: MangataAdapter | undefined = undefined;
+
+  beforeAll(async () => {
+    // Create keyringPair
+    keyringPair = await getKeyringPair();
+
+    // Get configs
+    const turingConfig = getOakConfig();
+    const mangataConfig = getMangataConfig();
+
+    // Initialize adapters
+    turingApi = await ApiPromise.create({ provider: new WsProvider(turingConfig.endpoint), rpc, types, runtime });
+    turingAdapter = new OakAdapter(turingApi, turingConfig);
+    await turingAdapter.initialize();
+
+    mangataSdk = Mangata.getInstance([mangataConfig.endpoint]);
+    mangataApi = await mangataSdk.getApi();
+    mangataAdapter = new MangataAdapter(mangataApi, mangataConfig);
+    await mangataAdapter.initialize();
+  }, 1000000);
+
+  test('test-turing-adapter', async () => {
+    if (!turingApi || !turingAdapter) throw new Error("Not initialized yet");
+    const extrinsic = turingApi.tx.system.remarkWithEvent('hello!');
+    const { encodedCallWeight, overallWeight } = await turingAdapter.getXcmWeight(extrinsic, '68TwNoCpyz1X3ygMi9WtUAaCb8Q6jWAMvAHfAByRZqMFEtJG', 6);
+    console.log('encodedCallWeight: ', encodedCallWeight);
+    console.log('overallWeight: ', overallWeight);
+    const { defaultAsset } = turingAdapter.getChainData();
+    const executionFee = await turingAdapter.weightToFee(overallWeight, defaultAsset?.location);
+    console.log('executionFee: ', executionFee.toString());
+  }, 1000000);
+
+  test('test-mangata-adapter', async () => {
+    if (!mangataApi || !mangataAdapter) throw new Error("Not initialized yet");
+    const extrinsic = mangataApi.tx.system.remarkWithEvent('hello!');
+    const { encodedCallWeight, overallWeight } = await mangataAdapter.getXcmWeight(extrinsic, '68TwNoCpyz1X3ygMi9WtUAaCb8Q6jWAMvAHfAByRZqMFEtJG', 6);
+    console.log('encodedCallWeight: ', encodedCallWeight);
+    console.log('overallWeight: ', overallWeight);
+    const executionFee = await mangataAdapter.weightToFee(overallWeight, assets.tur.location);
+    console.log('executionFee: ', executionFee.toString());
+  }, 1000000);
+
+  test('test-mangata-xcmp-task', async () => {
+    if (!mangataApi || !mangataAdapter || !turingAdapter || !keyringPair) throw new Error("Not initialized yet");
```

Comment:
!mangataAdapter || !turingAdapter || !keyringPair are not async jobs. Why is it necessary to check them here?

My understanding is that only turingApi and mangataApi needs initialization because they are async jobs.

Reply:
he TypeScript compiler enforces checking for these variables that could be 'undefined'.